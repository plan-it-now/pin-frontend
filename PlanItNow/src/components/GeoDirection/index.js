/* eslint no-console: 0 */
import React from 'react';
import getDirections from 'react-native-google-maps-directions';
import {
  StyleSheet,
  Text,
  View,
  Button
} from 'react-native';



class GeoDirection extends React.Component {

  constructor(props){
    super(props);
    this.state = {
      // initialPosition: 'unknown',
      lastPosition: 'unknown',
    };
  }


  watchID: ?number = null;

  componentDidMount() {
    // navigator.geolocation.getCurrentPosition(
    //   (position) => {
    //     var initialPosition = JSON.stringify(position);
    //     this.setState({initialPosition});
    //   },
    //   (error) => alert(JSON.stringify(error)),
    //   {enableHighAccuracy: true, timeout: 20000, maximumAge: 1000}
    // );
    this.watchID = navigator.geolocation.watchPosition((position) => {
      let lastPosition = JSON.stringify(position);
      this.setState({lastPosition});
    });
  }

  componentWillUnmount() {
    navigator.geolocation.clearWatch(this.watchID);
  }

  handleGetDirections = () => {
    const { myDestination } = this.props;
    const data = {
       source: {
        latitude: myLocation.latitude,
        longitude: myLocation.longitude
      },
      destination: {
        latitude: myDestination.latitude,
        longitude: myDestination.longitude
      },
      params: [
        {
          key: "dirflg",
          value: "w"
        }
      ]
    }

    getDirections(data)
  }

  render() {
    return (
        <View>
          <Button onPress={() => this.handleGetDirections()} title="Get Directions" />
        </View>
    );
  }
}


export default GeoDirection;
