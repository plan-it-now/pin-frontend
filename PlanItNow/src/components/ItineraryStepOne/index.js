import React from 'react';

import {
  View,
  Text,
  StyleSheet,
  ScrollView
} from 'react-native';

import {
  Container,
  Footer,
  FooterTab,
  Button,
  Content,
  Picker,
} from 'native-base';

import { connect } from 'react-redux';

const Item = Picker.Item;

class ItineraryStepOne extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      selectedItem: undefined,
      selected1: 'key1',
      results: {
        items: []
      }
    }
  }

  onValueChange ( value: String ) {
    this.setState({
      selected1: value
    })
  }

  render () {
    const { places } = this.props
    const arrPlaces = new Array(places.days);
    for (let i = 1; i <= arr.length; i++) {
      arr[i] = i;
    }
    return (
      <Container
        style={{ backgroundColor: '#B39DDB' }}
      >
        <ScrollView>
          <Content
            padder
          >
            <View
              style={{
                flex: 1,
                flexDirection: 'row',
                justifyContent: 'space-between',
                alignItems: 'center',
                backgroundColor: 'white',
                height: 80,
                marginBottom: 10,
                marginTop: 10
              }}
            >
              <View
                style={{
                  width: 200,
                  paddingLeft: 10
                }}
              >
                <Text
                  style={{
                    color: '#000',
                    fontSize: 18
                  }}
                >Pulau Dewata </Text>
              </View>
              <View style={{
                  width: 120,
                  paddingRight: 10
                }}
              >
                <Picker
                  supportedOrientations={['portrait','landscape']}
                  iosHeader="Select one"
                  mode="dropdown"
                  selectedValue={this.state.selected1}
                  onValueChange={this.onValueChange.bind(this)}>
                    {
                      arrPlaces.map(x => )
                    }
                </Picker>
              </View>
            </View>

          </Content>
        </ScrollView>


        <Footer>
          <FooterTab>
            <Button
              style={{ backgroundColor: '#5E35B1'}}
              block>
              <Text style={{ color: '#fff', fontWeight: 'bold' }}>Save</Text>
            </Button>
          </FooterTab>
        </Footer>
      </Container>
    )
  }
}

const mapStateToProps = state => ({
  places: state.places,
})
export default connect(mapStateToProps,null)(ItineraryStepOne);
