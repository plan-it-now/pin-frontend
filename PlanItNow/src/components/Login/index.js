import React from 'react';
import axios from 'axios';
import {
  View,
  Text,
  StyleSheet,
  Image
} from 'react-native';

import {
  Container,
  Content,
  Form,
  Item,
  Input,
  Label,
  Button,
  Icon
} from 'native-base';

class Login extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: '',
    }

  }

  static navigationOptions = {
    headerTitle: 'SignIn',
    headerTintColor: '#000',
    headerStyle: {
      backgroundColor: '#FFF59D'
    }

  };

  render () {
    return (
      <View
        style={styles.containerLogin}
      >
        <Image
          style={styles.backgroundImage}
          source={{ uri : 'http://4.bp.blogspot.com/-dXMb3OIPqiw/UV5pkcbDQhI/AAAAAAAACPw/HnPIlLoEbj0/s1600/3.jpg' }}

        >
          <View
            style={{
              width: '70%',
              height: '60%',
            }}>
            <Image
              style={{
                width: 80,
                height: 80,
                alignSelf: 'center',
              }}
              source={{ uri : 'http://www.freeiconspng.com/uploads/orange-location-icon-png-18.png'}}
            />
            <Item floatingLabel>
              <Icon name='person' style={{fontSize: 20, color: 'white'}} />

              <Label
                style={{color: '#fff'}}
              >Username</Label>
              <Input
                onChangeText = {(text) => this.setState({email:text})}
                style={{color: '#F44336'}}
              >

              </Input>
            </Item>
            <Item floatingLabel>
              <Icon name='lock' style={{fontSize: 20, color: 'white'}} />
              <Label
                style={{color: '#fff'}}
              >Password</Label>
              <Input
                onChangeText = {(text) => this.setState({password:text})}
              />
            </Item>
            <Button
                block warning
                style={{
                  marginTop: 20,
                  alignItems: 'center',
                  backgroundColor: '#FFEB3B'
                }}
                >
                <Text
                  style={{color: '#000'}}
                  >SignIn</Text>
              </Button>
              <Button
                  block warning
                  style={{
                    marginTop: 20,
                    alignItems: 'center',
                    backgroundColor: '#FFEB3B'
                  }}
                  >
                  <Text
                    style={{color: '#000'}}
                    >Signup</Text>
                </Button>
          </View>
        </Image>
    </View>
    )
  }

}

const styles = StyleSheet.create({
  containerLogin: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between'
  },
  backgroundImage: {
    flex: 1,
    zIndex: 100,
    resizeMode: 'cover',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  }
});

export default Login;
