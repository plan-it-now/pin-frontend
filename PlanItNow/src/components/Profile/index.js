import React from 'react';

import {
  View,
  Image,
  TouchableHighlight
} from 'react-native';

import {
  Container,
  Header,
  Title,
  Content,
  Footer,
  FooterTab,
<<<<<<< HEAD
  Button,
  Left,
  Right,
  Body,
  Card,
  CardItem,
  Text,
} from 'native-base';

import Icon from 'react-native-vector-icons/MaterialCommunityIcons';


=======
  Button, Left, Right, Body, Icon
} from 'native-base';

>>>>>>> resolve nothing to changes
class Profile extends React.Component {
  constructor(props) {
    super(props)

  }

  render () {
    return (
<<<<<<< HEAD
      <Container style={{ backgroundColor: '#B39DDB' }}>
        <Header style={{ backgroundColor: '#5E35B1' }}>
          <Left>
            <Button style={{
              backgroundColor: '#5E35B1',

            }}>
              <Icon name="menu" color="white" size={22}
              />
=======
      <Container>
        <Header>
          <Left>
            <Button transparant>
              <Icon name='menu' />
>>>>>>> resolve nothing to changes
            </Button>
          </Left>
          <Body>
            <Title>
              Your Name
            </Title>
          </Body>
        </Header>

<<<<<<< HEAD
        <Content padder>
          <Card>
            <View>
              <TouchableHighlight onPress={this._onPressButtoon}>
                <Image
                  style={{
                    flex: 1,
                    flexDirection: 'column',
                    justifyContent: 'flex-end',
                    width: '100%',
                    height: 200,
                    backgroundColor: 'rgba(52, 52, 52, 0.8)' }}
                  source={{ uri : 'http://alindstransport.com/wp-content/uploads/2016/09/Tugu-Jogja-Yogyakarta-jogja.jpg' }}>
                  <View style={{ flex: 1, flexDirection: 'row', justifyContent: 'space-between', alignItems: 'flex-end', backgroundColor: 'rgba(52, 52, 52, 0.4)' }}>
                    <Text style={{
                        color: '#fff', fontWeight: 'bold', margin: 5, flexWrap: 'wrap'
                      }}>Yogyakarta Trip</Text>
                    <Text style={{ color: '#fff', margin: 5}}>
                      12 Mei 2017
                    </Text>
                  </View>
                </Image>
              </TouchableHighlight>
              <CardItem>
                <Body>
                  <Text>
                    Tugu, Yogyakarta
                  </Text>
                  <Text note>
                    2 days ago
                  </Text>
                </Body>
              </CardItem>
            </View>

            <View>
              <Image
                style={{
                  flex: 1,
                  flexDirection: 'column',
                  justifyContent: 'flex-end',
                  width: '100%',
                  height: 200,
                  backgroundColor: 'rgba(52, 52, 52, 0.8)' }}
                source={{ uri : 'https://4.bp.blogspot.com/-4X3c0CA9Qp4/V-t1RvcGgII/AAAAAAAAAWY/CM_VOv8vjeYRpfnn6FJeIoZP44jQx6gSACLcB/s1600/600.PNG' }}>
                <View style={{ flex: 1, flexDirection: 'row', justifyContent: 'space-between', alignItems: 'flex-end', backgroundColor: 'rgba(52, 52, 52, 0.4)' }}>
                  <Text style={{
                      color: '#fff', fontWeight: 'bold', margin: 5, flexWrap: 'wrap'
                    }}>Bali Trip</Text>
                  <Text style={{ color: '#fff', margin: 5}}>
                    12 Mei 2017
                  </Text>
                </View>
              </Image>
              <CardItem>
                <Body>
                  <Text>
                    Tugu, Yogyakarta
                  </Text>
                  <Text note>
                    2 days ago
                  </Text>
                </Body>
              </CardItem>
            </View>

          </Card>
        </Content>

        <Button
          style={{
            justifyContent: 'center',
            width: 60,
            height: 60,
            backgroundColor: '#5E35B1',
            position: 'absolute',
            bottom: 10,
            right: 20,
            borderRadius: 100
          }}>
          <Icon name="plus" color="white" size={26} style={{
            textAlign: 'center',
            }}/>
        </Button>

=======
        <Content>
          <Text>Test</Text>
        </Content>

        <Footer backgroundColor="#fff">
          <Button block>
            <Text>Add</Text>
          </Button>
        </Footer>
>>>>>>> resolve nothing to changes
      </Container>
    )
  }

}

export default Profile;
