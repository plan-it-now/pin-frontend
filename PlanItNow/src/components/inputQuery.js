import React, { Component } from 'react';
import {
  Text,
  View
} from 'react-native';

import { Icon, Container, Col, Grid } from 'native-base';

const inputQuery = () => {
  return(
    <Container style={{flex:1}}>
      <View style={{marginTop:400}}>
        <Text style={{fontSize: 50}}>COBA</Text>
      </View>
    </Container>
  )
}

export default inputQuery
