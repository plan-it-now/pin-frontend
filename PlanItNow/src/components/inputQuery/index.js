import React, { Component } from 'react';
import {
  Text,
  View,
  BackHandler
} from 'react-native';
import { connect } from 'react-redux';
import { Container, Picker, Button, Form, Item as Stay, Input, Label, Header, Body, Title } from 'native-base';

import { fetchPlaces } from '../../actions';
const Item = Picker.Item;

class inputQuery extends Component {
  constructor(props){
    super(props)
    this.state = {
      city: 'key0',
      days:'',
      warning: ''
    }
    this.submitQuery = this.submitQuery.bind(this);
  }

  componentDidMount() {
    BackHandler.addEventListener('hardwareBackPress', this.handleBackButton)
  }

  componentWillUnmount() {
    BackHandler.removeEventListener('hardwareBackPress', this.handleBackButton)
  }

  handleBackButton() {
    return true
  }

  static navigationOptions = {
    headerTitle: "Inquiry"
  }

  onValueChange (value: string) {
    this.setState({
        city: value,
    });
  }

  submitQuery() {
    this.setState({warning:''})
    const { pref, fetchPlaces } = this.props;
    fetchPlaces(pref,this.state.city.toLowerCase());
    this.props.navigation.navigate('Recomendation');
  }


  render() {
    return(
      <Container style={{flex:1, backgroundColor:'#B39DDB'}}>
      <Header style={{backgroundColor:'#5E35B1'}}>
      <Body>
      <Title>Inquiry</Title>
      </Body>
      </Header>
        <View style={{flex:1, justifyContent:'center', flexDirection:'row', marginTop:50}}>
          <View style={{width: 300, flexDirection:'column'}}>
            <Picker
              style={{marginBottom: 50}}
                supportedOrientations={['portrait']}
                mode="dropdown"
                selectedValue={this.state.city}
                onValueChange={this.onValueChange.bind(this)}>
                <Item label="Location" value="key0" />
                <Item label="Semarang" value="Semarang" />
                <Item label="Yogyakarta" value="Yogyakarta" />
                <Item label="Bali" value="Bali" />
            </Picker>
            <Form>
              <Stay floatingLabel>
                <Label>Length of Stay</Label>
                <Input onChangeText={(days) => this.setState({days})}/>
              </Stay>
            </Form>
            <View style={{flex:1,justifyContent:'center', marginTop:50, flexDirection:'row'}}>
              <Button rounded warning onPress={() => {(this.state.days !== '' && this.state.city !== 'key0') ? this.submitQuery() : this.setState({warning:'Please input all fields'})}}>
                 <Text>Lets Go!</Text>
             </Button>
             <Text>{this.state.warning}</Text>
           </View>
          </View>
        </View>

      </Container>
    )
  }
}

const mapStateToProps = state => ({
  pref: state.logindata.userdata.pref
})

const mapDispatchToProps = dispatch => ({
  fetchPlaces: (pref,city) => dispatch(fetchPlaces(pref,city))
})

export default connect(mapStateToProps,mapDispatchToProps)(inputQuery);
