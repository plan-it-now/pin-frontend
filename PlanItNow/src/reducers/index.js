import { combineReducers } from 'redux';

import userReducer from './userReducer';
import placeReducer from './placeReducer';
import itineraryReducer from './itineraryReducer';

const rootReducer = combineReducers({
  logindata: userReducer,
  places: placeReducer,
  itineraries: itineraryReducer
});

export default rootReducer;
