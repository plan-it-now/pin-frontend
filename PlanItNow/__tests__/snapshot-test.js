import 'react-native';
import React from 'react';
import PlanItNow from '../index.android.js';
import { shallow } from  'enzyme';
import reducers from '../src/reducers';

import { Provider } from 'react-redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import { applyMiddleware, createStore } from 'redux';

import Login from '../src/Components/Login';
import Profile from '../src/Components/Profile';
import GeoDirection from '../src/Components/GeoDirection';
import inputQuery from '../src/Components/inputQuery';
import ItineraryDetail from '../src/Components/ItineraryDetail';
import ItineraryStepOne from '../src/Components/ItineraryStepOne';
import MyComponent from '../src/Components/step2';

const store = createStore(reducers, applyMiddleware(thunk, logger));

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('Should render Login Component', () => {
  const LoginComp = shallow(
    <Provider store={store}>
      <Login />
    </Provider>
  );
  expect(LoginComp).toMatchSnapshot();
})

it('Should render Profile Component', () => {
  const profileComp = shallow(
    <Provider store={store}>
      <Profile />
    </Provider>
  );
  expect(profileComp).toMatchSnapshot();
})

it('Should render GeoDirection Componet', () => {
  const GeoDirectionComp = shallow(
    <Provider store={store}>
      <GeoDirection />
    </Provider>
  );
  expect(GeoDirectionComp).toMatchSnapshot();
})

it('Should render inputQuery Componet', () => {
  const inputQueryComp = shallow(
    <Provider store={store}>
      <inputQuery />
    </Provider>
  );
  expect(inputQueryComp).toMatchSnapshot();
})

it('Should render ItineraryStepOne Componet', () => {
  const ItineraryStepOneComp = shallow(
    <Provider store={store}>
      <ItineraryStepOne />
    </Provider>
  );
  expect(ItineraryStepOneComp).toMatchSnapshot();
})

it('Should render MyComponent Componet', () => {
  const MyComponentComp = shallow(
    <Provider store={store}>
      <MyComponent />
    </Provider>
  );
  expect(MyComponentComp).toMatchSnapshot();
})
