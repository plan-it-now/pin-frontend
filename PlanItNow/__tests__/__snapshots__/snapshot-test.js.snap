// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render GeoDirection Componet 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <GeoDirection />,
  "nodes": Array [
    <GeoDirection />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        }
>
        <GeoDirection />
</Provider>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <GeoDirection />,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <GeoDirection />,
        "_debugID": 6,
        "_renderedOutput": <GeoDirection />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
>
    <GeoDirection />
</Provider>,
}
`;

exports[`Should render ItineraryStepOne Componet 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(ItineraryStepOne) />,
  "nodes": Array [
    <Connect(ItineraryStepOne) />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        }
>
        <Connect(ItineraryStepOne) />
</Provider>,
      "_debugID": 9,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <Connect(ItineraryStepOne) />,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 5,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(ItineraryStepOne) />,
        "_debugID": 10,
        "_renderedOutput": <Connect(ItineraryStepOne) />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
>
    <Connect(ItineraryStepOne) />
</Provider>,
}
`;

exports[`Should render Login Component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(Login) />,
  "nodes": Array [
    <Connect(Login) />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        }
>
        <Connect(Login) />
</Provider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <Connect(Login) />,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(Login) />,
        "_debugID": 2,
        "_renderedOutput": <Connect(Login) />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
>
    <Connect(Login) />
</Provider>,
}
`;

exports[`Should render MyComponent Componet 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(MyComponent) />,
  "nodes": Array [
    <Connect(MyComponent) />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        }
>
        <Connect(MyComponent) />
</Provider>,
      "_debugID": 11,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <Connect(MyComponent) />,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 6,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(MyComponent) />,
        "_debugID": 12,
        "_renderedOutput": <Connect(MyComponent) />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
>
    <Connect(MyComponent) />
</Provider>,
}
`;

exports[`Should render Profile Component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(Profile) />,
  "nodes": Array [
    <Connect(Profile) />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        }
>
        <Connect(Profile) />
</Provider>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <Connect(Profile) />,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(Profile) />,
        "_debugID": 4,
        "_renderedOutput": <Connect(Profile) />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
>
    <Connect(Profile) />
</Provider>,
}
`;

exports[`Should render inputQuery Componet 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <inputQuery />,
  "nodes": Array [
    <inputQuery />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        }
>
        <inputQuery />
</Provider>,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <inputQuery />,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <inputQuery />,
        "_debugID": 8,
        "_renderedOutput": <inputQuery />,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
>
    <inputQuery />
</Provider>,
}
`;
